examples/quad/main.rs

```diff
@@ -137,663 +138,751 @@ fn main() {
         println!("{:?}", adapter.info);
     }
 
-    let mut adapter = adapters.remove(0);
-    let memory_types = adapter.physical_device.memory_properties().memory_types;
-    let limits = adapter.physical_device.limits();
+    let adapter = adapters.remove(0);
 
-    // Build a new device and associated command queues
-    let (device, mut queue_group) = adapter
-        .open_with::<_, hal::Graphics>(1, |family| surface.supports_queue_family(family))
-        .unwrap();
+    let mut renderer = Renderer::new(surface, adapter);
 
-    let mut command_pool = unsafe {
-        device.create_command_pool_typed(&queue_group, pool::CommandPoolCreateFlags::empty())
-    }
-    .expect("Can't create command pool");
+    // It is important that the closure move captures the Renderer,
+    // otherwise it will not be dropped when the event loop exits.
+    event_loop.run(move |event, _, control_flow| {
+        *control_flow = winit::event_loop::ControlFlow::Wait;
 
-    // Setup renderpass and pipeline
-    let set_layout = unsafe {
```

Comment:
why are we even touching all this code? Creating the resources doesn't depend on winit, only the swapchain is.